### PR TITLE
feat: modal.update() supports functional updating (#27162)

### PR DESCRIPTION
**Pull request by Mongkii**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

Related issue: #27162 

### 💡 Background and solution

Story: [CodeSandbox](https://codesandbox.io/s/friendly-breeze-mkthf?file=/src/App.tsx)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | `modal.update()` supports functional updating |
| 🇨🇳 Chinese |`modal.update()` 支持函数式更新 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/modal/index.en-US.md](https://github.com/Mongkii/ant-design/blob/feature/components/modal/index.en-US.md)
[View rendered components/modal/index.zh-CN.md](https://github.com/Mongkii/ant-design/blob/feature/components/modal/index.zh-CN.md)

### CODE REVIEW

**Review by MrHeer — APPROVED** [review]
LGTM

**Comment by hengkx** [comment]
please add api version

File: components/modal/index.zh-CN.md [hunk 1]
```diff
@@ -99,6 +99,11 @@ modal.update({
   content: '修改的内容',
 });
 
+modal.update(prevConfig => ({
```
Comment (hengkx) [inline components/modal/index.zh-CN.md:102]: please add api version

**Review by hengkx — CHANGES_REQUESTED** [review]
please add api version

File: components/modal/index.zh-CN.md (hunk 1 above)
Comment (Mongkii) [reply components/modal/index.zh-CN.md:102]: What version should I use? Is `4.7.3` ok?

Comment (hengkx) [reply components/modal/index.zh-CN.md:102]: 4.8.0

Comment (Mongkii) [reply components/modal/index.zh-CN.md:103]: Docs updated in f971a718

**Review by hengkx — APPROVED** [review]

**Pull request merged**